core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -716,4 +717,9 @@ default boolean isMaterializedView(Session session, QualifiedObjectName viewName
      * Returns a table handle for the specified table name with a specified version
      */
     Optional<TableHandle> getTableHandle(Session session, QualifiedObjectName tableName, Optional<TableVersion> startVersion, Optional<TableVersion> endVersion);
+
+    /**
+        Returns maximal number of tasks that should be in writing to specific connector.
+     */
+    OptionalInt getMaxWriterTasks(Session session, String catalogName);
```

Comment:
Instead of having a separate method, can we put this variable inside `ConnectorTableLayout`?

Reply:
What do you mean exactly?

Reply:
Ok, I thought about it a bit more and it makes sense to have a separate method since this functionality doesn't overlap with `ConnectorTableLayout`.

`ConnectorTableLayout` primarily describes the actual partitioning and related properties of the table whereas this is more related to remote partitioning which limits the number of writers.

WDYT? @sopel39